src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -619,28 +619,17 @@ private static bool IsFloatingPointExpressionOfUnknownPrecision(BoundExpression
                     {
                         var spanType = (NamedTypeSymbol)rewrittenType;
 
-                        WellKnownMember member;
-                        if (spanType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions))
-                        {
-                            member = WellKnownMember.System_ReadOnlySpan_T__op_Implicit_Array;
-                        }
-                        else
-                        {
-                            Debug.Assert(spanType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_Span_T), TypeCompareKind.AllIgnoreOptions));
-                            member = WellKnownMember.System_Span_T__op_Implicit_Array;
-                        }
-
-                        if (!TryGetWellKnownTypeMember(rewrittenOperand.Syntax, member, out MethodSymbol? symbol))
+                        if (Binder.TryFindImplicitOperatorFromArray(spanType.OriginalDefinition) is not { } methodDefinition)
                         {
                             throw ExceptionUtilities.Unreachable();
                         }
                         else
                         {
-                            MethodSymbol method = symbol.AsMember(spanType);
+                            MethodSymbol method = methodDefinition.AsMember(spanType);
 
                             rewrittenOperand = _factory.Convert(method.ParameterTypesWithAnnotations[0].Type, rewrittenOperand);
 
-                            if (member == WellKnownMember.System_ReadOnlySpan_T__op_Implicit_Array)
+                            if (_compilation.IsReadOnlySpanType(spanType))
```

Comment:
>_compilation.IsReadOnlySpanType(spanType) [](http://example.com/codeflow?start=32&length=41)

This uses `WellKnownMember.System_ReadOnlySpan_T` rather than checking by fully-qualified type name. Is that correct?

Reply:
That was my intent - I even have a test for it: `Conversion_Array_Span_Implicit_ConstantData_NotWellKnownSpan`. The emit of constant data for ReadOnlySpan (i.e., emit of `BoundReadOnlySpanFromArray`) expects the well-known span type and I don't think changing that is worth the effort.